pkg/operator/apis/aro.openshift.io/v1alpha1/cluster_types.go

```diff
@@ -45,25 +45,26 @@ type InternetCheckerSpec struct {
 // ClusterSpec defines the desired state of Cluster
 type ClusterSpec struct {
 	// ResourceID is the Azure resourceId of the cluster
-	ResourceID          string              `json:"resourceId,omitempty"`
-	Domain              string              `json:"domain,omitempty"`
-	ACRDomain           string              `json:"acrDomain,omitempty"`
-	AZEnvironment       string              `json:"azEnvironment,omitempty"`
-	Location            string              `json:"location,omitempty"`
-	InfraID             string              `json:"infraId,omitempty"`
-	ArchitectureVersion int                 `json:"architectureVersion,omitempty"`
-	GenevaLogging       GenevaLoggingSpec   `json:"genevaLogging,omitempty"`
-	InternetChecker     InternetCheckerSpec `json:"internetChecker,omitempty"`
-	VnetID              string              `json:"vnetId,omitempty"`
-	APIIntIP            string              `json:"apiIntIP,omitempty"`
-	IngressIP           string              `json:"ingressIP,omitempty"`
+	ResourceID             string              `json:"resourceId,omitempty"`
+	ClusterResourceGroupID string              `json:"clusterResourceGroupId,omitempty"`
+	Domain                 string              `json:"domain,omitempty"`
+	ACRDomain              string              `json:"acrDomain,omitempty"`
+	AZEnvironment          string              `json:"azEnvironment,omitempty"`
+	Location               string              `json:"location,omitempty"`
+	InfraID                string              `json:"infraId,omitempty"`
+	ArchitectureVersion    int                 `json:"architectureVersion,omitempty"`
+	GenevaLogging          GenevaLoggingSpec   `json:"genevaLogging,omitempty"`
+	InternetChecker        InternetCheckerSpec `json:"internetChecker,omitempty"`
+	VnetID                 string              `json:"vnetId,omitempty"`
+	APIIntIP               string              `json:"apiIntIP,omitempty"`
+	IngressIP              string              `json:"ingressIP,omitempty"`
 
 	Features FeaturesSpec `json:"features,omitempty"`
 }
 
 // FeaturesSpec defines ARO operator feature gates
 type FeaturesSpec struct {
-	ReconcileNSGs                  bool `json:"reconcileNSGs,omitempty"`
```

Comment:
Recommend we keep `reconcileNSGs` and deprecate unless we know for sure that it's not used anywhere.  The PR could include a migration of values from `reconcileNSGs` to `reconcileSubnets` but without this behavior in clusters will be impacted.

Reply:
@Makdaam how we are dealing with this controller now in prod?

Reply:
Confirmed with @Makdaam , we only few enabled and there was some fallback. So its safe to do `false` as we gonna revisit this in prod.